packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2.ts

```diff
@@ -53,21 +58,21 @@ export interface AmazonPayV2HostWindow extends Window {
     amazon?: AmazonPayV2SDK;
 }
 
-interface AmazonPayV2ButtonConfig {
+export interface AmazonPayV2ButtonConfig {
```

Comment:
Not sure about making all props optional because, according to Amazon docs, they are not :/

Reply:
@mauricio-sg when we initialize the strategy for **Buy Now** flow we don't have options and we collect the amazon config via this `_getAmazonPayV2ButtonOptions` method

You can see it there [L129](https://github.com/bigcommerce/checkout-sdk-js/blob/562ab380046cec191067d5c68e8f2dde221aab5c/packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts#L129)

We only have `options.buyNowInitializeOptions` witch we get from a script is loaded from bcapp
I can check if we can pass required parameters for Amazon from there but i think it is redundant because we get all these variables in `_getAmazonPayV2ButtonOptions` method

Reply:
Makes sense. How about modifying instead `packages/core/src/checkout-buttons/strategies/amazon-pay-v2/amazon-pay-v2-button-options.ts` and doing something like this:

```js
interface WithBuyNowFeatures {
    /**
     * The options that are required to initialize Buy Now functionality.
     */
    buyNowInitializeOptions?: {
        getBuyNowCartRequestBody?(): BuyNowCartRequestBody | void;
    };
}

/**
 * The required config to render the AmazonPayV2 buttton.
 */
export type AmazonPayV2ButtonInitializeOptions = AmazonPayV2ButtonParameters | WithBuyNowFeatures;
```

Reply:
@mauricio-sg thanks a lot for this example. I updated and added **type guard**. Take a look please